pkg/scheduler/framework/plugins/volumerestrictions/volume_restrictions.go

```diff
@@ -321,15 +328,66 @@ func (pl *VolumeRestrictions) EventsToRegister() []framework.ClusterEventWithHin
 		// Pods may fail to schedule because of volumes conflicting with other pods on same node.
 		// Once running pods are deleted and volumes have been released, the unschedulable pod will be schedulable.
 		// Due to immutable fields `spec.volumes`, pod update events are ignored.
-		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}},
+		{Event: framework.ClusterEvent{Resource: framework.Pod, ActionType: framework.Delete}, QueueingHintFn: pl.isSchedulableAfterPodDeleted},
 		// A new Node may make a pod schedulable.
+		// We intentionally don't set QueueingHint since all Node/Add events could make Pods schedulable.
 		{Event: framework.ClusterEvent{Resource: framework.Node, ActionType: framework.Add}},
 		// Pods may fail to schedule because the PVC it uses has not yet been created.
 		// This PVC is required to exist to check its access modes.
 		{Event: framework.ClusterEvent{Resource: framework.PersistentVolumeClaim, ActionType: framework.Add | framework.Update}},
 	}
 }
 
+// isSchedulableAfterPodDeleted is invoked whenever a pod deleted,
+// It checks whether the deleted pod will conflict with volumes of other pods on the same node
+func (pl *VolumeRestrictions) isSchedulableAfterPodDeleted(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	deletedPod, _, err := util.As[*v1.Pod](oldObj, newObj)
+	if err != nil {
+		return framework.Queue, fmt.Errorf("unexpected objects in isSchedulableAfterPodDeleted: %w", err)
+	}
+
+	if deletedPod.Namespace != pod.Namespace {
+		return framework.QueueSkip, nil
+	}
+
+	newPodPvcs, err := pl.readWriteOncePodPVCsForPod(pod, false)
```

Comment:
Our strategy of implementing QHint got a bit changed:
- _For now_, we try not to refer to other kinds of resources.
  - e.g., in this case, `isSchedulableAfterPodDeleted` only refers a unscheduled Pod and a deleted Pod. We shouldn't refer to PVC.
  - "_For now_" > Once we get a better observability around requeueing (https://github.com/kubernetes/kubernetes/issues/124566), we may again allow QHints to refer to other kinds of resources depending on the throughput degradation from them.
- If we cannot do any filtering with this strategy, we don't implement QHint for the event.

The reason is basically the throughput concern. The more complex each QHint becomes, the longer time each QHint consumes and we could get a scheduling throughput issue.


So, can you modify this QHint so that it becomes not-depending on PVC or other kinds of resources?

Reply:
Removed code referencing pvc and added a TODO stating that we will revert the check for pvc once we observe good throughput.